openmrs/omod/src/main/java/org/openmrs/projectbuendia/webservices/rest/RequestUtil.java

```diff
@@ -0,0 +1,40 @@
+// Copyright 2015 The Project Buendia Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License.  You may obtain a copy
+// of the License at: http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distrib-
+// uted under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+// OR CONDITIONS OF ANY KIND, either express or implied.  See the License for
+// specific language governing permissions and limitations under the License.
+
+package org.openmrs.projectbuendia.webservices.rest;
+
+import org.openmrs.module.webservices.rest.web.RequestContext;
+import org.openmrs.projectbuendia.Utils;
+
+import javax.annotation.Nullable;
+import java.text.ParseException;
+import java.util.Date;
+
+/**
+ * Utilities for working with requests and request parameters.
+ */
+public class RequestUtil {
+    /**
+     * Obtains the "Sync From" date from a request. This is stored in the "since" HTTP parameter
+     * in ISO 8601 format.
+     * @param context the relevant request.
+     * @return the instant that data should be provided from (inclusive), or {@code null} if the
+     * "since" parameter was not present in the request.
+     */
+    @Nullable
+    public static Date getSyncFromDate(RequestContext context) throws ParseException {
+        String param = context.getParameter("since");
+        if (param == null) {
+            return null;
+        }
+        return Utils.fromIso8601(param);
```

Comment:
Okay.  This is definitely your call (and honestly I can see your side of it
too).
On Nov 2, 2015 9:47 PM, "Fabian Tamp" <notifications@github.com> wrote:

> In
> openmrs/omod/src/main/java/org/openmrs/projectbuendia/webservices/rest/RequestUtil.java
> <https://github.com/projectbuendia/buendia/pull/20#discussion_r43718008>:
>
> > + */
> > +public class RequestUtil {
> > +    /**
> > +     * Obtains the "Sync From" date from a request. This is stored in the "since" HTTP parameter
> > +     * in ISO 8601 format.
> > +     * @param context the relevant request.
> > +     * @return the instant that data should be provided from (inclusive), or {@code null} if the
> > +     * "since" parameter was not present in the request.
> > +     */
> > +    @Nullable
> > +    public static Date getSyncFromDate(RequestContext context) throws ParseException {
> > +        String param = context.getParameter("since");
> > +        if (param == null) {
> > +            return null;
> > +        }
> > +        return Utils.fromIso8601(param);
>
> Millis can still suffer from timezone problems though - you're just hiding
> it by rolling it into the actual timestamp. I'm going to keep this as ISO
> 8601; it's easier to debug (which is a huge plus for me given how much
> request debugging I'll be doing in the coming weeks [image: :wink:] ),
> and seeing as it's treated as a token by the client (i.e. no need to parse
> it), we don't have to worry as much about compatibility - we're only
> working with a single implementation.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/projectbuendia/buendia/pull/20/files#r43718008>.
>